### DISCUSSION TITLE
CP2K Compilation

### DISCUSSION THREAD

**Discussion by gcarneiroq** [mir-group/allegro#73] [Q&A]
Hi, folks.

I am very interested in starting to learn ALLEGRO and my idea was to give a shot with CP2K. However, i am struggling for compiling the code with the libtorch.

I tried to follow the instructions in the CP2K manual: https://manual.cp2k.org/trunk/methods/machine_learning/nequip.html . 

However using the toolchain doesnt work. I tried different ways.

1. just following the --with-libtorch=install.
The problem here seems that the toolchain downloads from a static repository of CP2K, the version of libtorch it gets it is one without gpu support. However, the source files (specifically: torch_c_api.cpp) points to a cuda header. 

2. So I downloaded the libtorch with gpu support that contains the aforementioned header. However, the cp2k installation fails to get the correct libtorch header paths.

3. I tried different ways to point towards the right path for include folders: exporting environment variables, compiling libtorch from scratch, compiling with python, compiling with cmake... nothing  seems to generate the correct environment variables. The only way I manage to advance the CP2K compilation with libtorch was to manually edit the headers, however I gave up. 

So i admit I have zero knowledge of pytorch and, therefore, libtorch, so I am asking for a guidance on how to compile CP2K with libtorch or to correctly set the Path for the include variables of libtorch so CP2K can find it.

Thank you for your attention.

Kind regards,

Guilherme

**Comment by gcarneiroq** [comment]
Some updates I got with the issue.

I tried to ask toolchain to compile the gcc-compiler and this seems to have solved the problem:

./install_cp2k_toolchain.sh --with-gcc=install --with-spglib=no --with-cosma=no --with-libxsmm=no --with-libvori=no --with-libxc=no --with-plumed=no --with-sirius=no --with-libint=no --with-fftw=no --with-libvdwxc=no --with-gsl=no --with-elpa=no --with-libtorch --with-intel=no --with-intelmpi=no

It was an ubuntu 22.04 and the care that I really needed to take was to be sure inteloneapi suite was not loaded. From here I will try to compile with more features to see if everything works.

**Reply by Linux-cpp-lisp** [reply]
Hi @gcarneiroq ,

Thanks for your interest in Allegro.  You may have more luck asking the CP2K developers, who wrote this integration and have successfully built it in the past.  In our experience with LAMMPS, building `libtorch` from source can be necessary on many machines, but it sounds like you've already tried that.

Generally, if you do not need CP2K specific features, you could also try our LAMMPS integration as your MD engine. (See the `pair_allegro` repository.)

**Comment by Linux-cpp-lisp** [comment]
Closing as outdated; the current LAMMPS integration and more information on it can be found at https://github.com/mir-group/pair_nequip_allegro.
